### DISCUSSION TITLE
[FEATURE] ARG help

### DISCUSSION THREAD

**Discussion by amuilyas** [Tzahi12345/YoutubeDL-Material#405] [Q&A]
First of all thank you for such a great program which I find incredibly useful for watching videos offline at my own pace and with full playback control on Kodi.

**Is your feature request related to a problem? Please describe.**
I find the documentation on adding arguments confusing.  I am definitely NOT a programmer but like to think of myself as semi-intelligent and struggled to follow the guidelines with respect to the GUI vs CLI that the docs seem to assume.

**Describe the solution you'd like**
One or two examples in the docs that show the text string within the GUI as a screenshot.  My immediate problem is wanting to add file creation/upload date at the start of the file name so I can easily sort channel downloads by date for latest videos.  I tried several permutations of arg text based on what I could google but do not get dates at start of file.

I tried below and none worked for me:
-o"%(release_date)s - %(title)s.%(ext)s"
-o "%(release_date)s - %(title)s.%(ext)s"
-o '%(release_date)s.%(title)s.%(ext)s'

**Additional context**
I am running YDL on Synology in Docker.  Picture of my failed attempts.  I know I am doing something very wrong, just need some pointers on how do it properly.  Thanks again for a great program!
![Screenshot 2021-07-24 182058](https://user-images.githubusercontent.com/77816893/126884743-52df39d3-5437-4890-ad52-09aacb3271ed.png)

**Comment by Tzahi12345** [comment]
Will convert this to a discussion as it's more of a question, but I'll answer it there!

**Comment by Tzahi12345** [comment]
There is a better place to add the output arg specifically, but I'll explain how the args menu works (scroll down for the better solution):

I agree this is a bit confusing! You're close though, the top card is the actual args list and the bottom one is a place to add them, though they kind of both serve as places to add args.

Anyways, if you want to do this in the top card, just type `-o` then a comma, which should create a little bubble around the `-o`. Then you can type your custom output `%(release_date)s - %(title)s.%(ext)s`, even with spaces. After you're done with that, type in another comma to create a bubble around the desired output string. It should look like this in the end:

![image](https://user-images.githubusercontent.com/707413/127074845-a38dfea1-4dda-4fb8-b6e3-a5850e250f45.png)

The bottom card is just a little tool, if you were to do it there you would type in `-o` in the `Arg` input, then click on the `Use arg value` checkbox and type in the output string in the `Arg value` input. Then click add and it would be added in one go.

--------------

Now for the more robust way to add a custom output is actually to just use the `Default file output` setting like so:

![image](https://user-images.githubusercontent.com/707413/127075573-a38a0607-1157-4174-9a5f-56a7ba20c187.png)

Notice how the `.%(ext)s` is missing: it's not required!

Hope this makes sense, sorry for the huge wall of text. But custom args can be very powerful and I wanted to make sure you weren't lost on that either.

**Reply by amuilyas** [reply]
Appreciate the reply!

I tried the global settings and do not see the change in the file name.  Not sure if related but there is a JSON error in the log file.
- deleted existing subscriptions
- turned YTDL off in docker
- deleted the channel folders in NAS to start from blank slate
- turned YTDL on in docker
- added 3 different channels to subs with low video count uploads

Thanks again for helping me out with this and big fan of YTDL

![1 YTDL screenshot](https://user-images.githubusercontent.com/77816893/127242953-13826cfc-d9b3-44f1-bdc6-1c9c2dc90c43.png)

![2 NAS screenshot](https://user-images.githubusercontent.com/77816893/127242959-197f693a-33d5-4dae-84cd-5b1caa516a7a.png)

![3 Docker screenshot](https://user-images.githubusercontent.com/77816893/127242961-d91fe996-a80b-4b10-bc67-879ced0e176d.png)

**Reply by Tzahi12345** [reply]
Huh, it worked on my end:

![image](https://user-images.githubusercontent.com/707413/127263847-94005927-065f-4fea-bfa0-fbd24cc00bcd.png)

Are these subscription videos? If so, I should've mentioned this earlier but the default file output only applies to non-subscription videos. At least it did, I just pushed [an update](https://github.com/Tzahi12345/YoutubeDL-Material/commit/c0a385ce78f0e5c4edca6acc2ed6566c5bc3701d) that fixes this,

The JSON error though is strange, would you mind providing the full logs? You can find this in the settings menu, there's a `Logs` tab.

**Reply by amuilyas** [reply]
Thanks mate!

My end game is to use YTDL to download videos to watch offline in media player.  But the media player will only alpha sort by file name so I want to append creation date to the start of the file names so I can embed a time based sort.  All of the channels are non-subscription.

I added my log file here for your reference.

I notice in your screenshot that the file has been appended with "NA - " at the beginning.  Am I right that in your example it does not seem to have edited the file name with creation date?

I am running v4.2 of YTDL in Docker, it seems to have grabbed your most recent file update.

thanks again for the support
[YTDL_log.txt](https://github.com/Tzahi12345/YoutubeDL-Material/files/6896204/YTDL_log.txt)

**Reply by Tzahi12345** [reply]
Hm so it seems like an apostrophe was added for whatever reason to the expected name. But, we were getting the file name using a rather unsafe method, which has since been changed to something a lot more robust which I'm almost positive will fix your issue.

You'll have to change your tag to `nightly` and join us in a more bleeding edge version of YTDL-M. You may want to back up your instance as there's a DB migration involved, but it should be relatively straightforward in your case and the build seems relatively stable. One added benefit is your subscriptions will have `%(release_date)s` added to their output which you seem to want as well. Let me know if you need more help on this front.

>I notice in your screenshot that the file has been appended with "NA - " at the beginning. Am I right that in your example it does not seem to have edited the file name with creation date?

It seems like youtube-dl didn't put that info there for whatever reason. Not sure why.

**Comment by amuilyas** [comment]
Thanks for that.



I'll wait until you push the changes out on to a stable version.



Big fan of YTDL!





---- On Thu, 29 Jul 2021 20:32:16 -0700 Tzahi12345 ***@***.***> wrote ----




Hm so it seems like an apostrophe was added for whatever reason to the expected name. But, we were getting the file name using a rather unsafe method, which has since been changed to something a lot more robust which I'm almost positive will fix your issue.

You'll have to change your tag to nightly and join us in a more bleeding edge version of YTDL-M. You may want to back up your instance as there's a DB migration involved, but it should be relatively straightforward in your case and the build seems relatively stable. Let me know if you need more help on this front.

I notice in your screenshot that the file has been appended with "NA - " at the beginning. Am I right that in your example it does not seem to have edited the file name with creation date?


It seems like youtube-dl didn't put that info there for whatever reason. Not sure why.
